README.md

```diff
@@ -98,10 +100,11 @@ $ ttn-lw-migrate devices --source ttnv2 < device_ids.txt > devices.json
 Similarly, to export all devices of application `my-app-id`:
 
 ```bash
+$ export TTNV2_APP_ID="my-app-id"                     # TTN App ID
 # dry run first, verify that no errors occur
-$ ttn-lw-migrate application --source ttnv2 "my-app-id" --dry-run --verbose > devices.json
+$ ttn-lw-migrate application --source ttnv2 --dry-run --verbose > devices.json
 # export devices
-$ ttn-lw-migrate application --source ttnv2 "my-app-id" > devices.json
+$ ttn-lw-migrate application --source ttnv2 > devices.json
```

Comment:
Maybe I don't really get the problem, but I don't find this improving the user experience.

Args should take precedence or we shouldn't take args.

> TTNv2: the args are ignored since the appID is read from env. So no matter how many args a provided, it only migrates the one single application read from env. This is because the v2 manager client works on single applications.

If that is the case, why would one put the application ID in the env, and not always use an argument? Also for device export?

---

Leaving ChirpStack aside, for TTNV2 migration, why not;

Command | Scope
--- | ---
[1] `ttn-lw-migrate application --source ttnv2 app1` | application
[2] `ttn-lw-migrate devices --source ttnv2 app1 < devices.txt` | multiple devices
[3] `ttn-lw-migrate devices --source ttnv2 app1 device1` | one device

Or am I missing something?

Is the problem that we have the following:

Command | Scope
--- | ---
[4] `ttn-lw-migrate devices --source ttnv2 device1` | one device

And that we cannot disambiguate between [2] and [4]? We can check if there's a reader, right? See https://github.com/TheThingsIndustries/lorawan-stack/blob/v3.16/cmd/internal/io/io.go#L94-L105. Or should we not support [4] and require [3] instead?

---

This `dummy` iterator thing degrades the quality of the codebase. This shouldn't be necessary. If we need to break command-line args